MapboxCoreNavigation/RouteController.swift

```diff
@@ -767,6 +784,15 @@ extension RouteController: CLLocationManagerDelegate {
         #endif
     }
 
+    private func checkForLocationUsageDescription() {
+        guard let _ = Bundle.main.bundleIdentifier else {
+            return
+        }
+        if Bundle.main.locationAlwaysUsageDescription == nil && Bundle.main.locationWhenInUseUsageDescription == nil && Bundle.main.locationAlwaysAndWhenInUseUsageDescription == nil {
```

Comment:
Nit: This is fine, but I find it more natural-reading to use commas as opposed to boolean-and's in swift conditionals when the conditional is a simple "if all this stuff is T/F". 

`if thisThing, thatThing, otherThing {` 

vs 

`if thisThing && thatThing && otherThing {`

Reply:
@JThramer Although, it's an elegant approach, being explicit can also help distinguish where the value could be a boolean value or optional. An explicit definition in this conditional statement helps specify our main goal here. Yes?

Reply:
@vincethecoder I'm not sure I understand what you're getting at, the `locationAlwaysUsageDescription == nil` is required in either case, as `locationAlwaysUsageDescription` is not a `Bool`.

`,` is literally just conditional syntactic sugar for `&&`. It works in any conditional where `&&` does. IMHO, boolean and's are appropriate when you have some sort of complex conditional logic, e.g: `if thisThing && thatThing || thatOtherThing` or `(thisThing && thatThing) || orThisThing`, but if you are just checking "all these clauses in this expression are true", then it's more natural-reading to use commas to join the clauses of this expression.

Reply:
1) this code was just moved from init
2) this code arguably doesn't need/deserve to exist at all, but we're minimizing changes at this stage in the game

Reply:
@JThramer I was inferring when there are instances where a field could be `Bool?`, then we should be explicit. Or there is a series of `Bool` and `Optional<Nil>` values in one conditional statement. 

It's just self-documenting. Yes, we can also short-key click to inspect the variable. In this thread, I just said it is alright to be explicit. It might be additional LOC but it adds clarity to the code -- that's all.